test/modify-status.spec.js

```diff
@@ -0,0 +1,133 @@
+						);
+					});
+					describe('And the WFH contains an emoji followed by a space and a status message', () => {
+						beforeEach(() => {
+							status.status_text = text;
+							status.status_emoji = ':'.concat(emoji.which(randomEmoji), ':');
+							text = randomEmoji.concat(' ', text);
+							return act();
+						});
+						it('should set status emoji and message to supplied text, excluding space between emoji and message', () =>
+							sinon.assert.calledWith(fakeSlackApi.sendResponse, 'slack.com/api/users.profile.set', 'token='.concat(token, '&profile=', encodeURIComponent(JSON.stringify(status))))
+						);
+
+					});
+					describe('And the WFH contains an emoji followed immediately by a status message', () => {
+						beforeEach(() => {
+							status.status_text = text;
+							status.status_emoji = ':'.concat(emoji.which(randomEmoji), ':');
+							text = randomEmoji.concat(text);
+							return act();
+						});
+						it('should set status emoji and message to supplied text', () =>
+							sinon.assert.calledWith(fakeSlackApi.sendResponse, 'slack.com/api/users.profile.set', 'token='.concat(token, '&profile=', encodeURIComponent(JSON.stringify(status))))
+						);
+					});
+					describe('And the WFH contains a message with no emoji', () => {
+						beforeEach(() => {
+							status.status_text = text;
+							return act();
+						});
+						it('should set status message to supplied text and status emoji to default :house_with_garden:', () =>
+							sinon.assert.calledWith(fakeSlackApi.sendResponse, 'slack.com/api/users.profile.set', 'token='.concat(token, '&profile=', encodeURIComponent(JSON.stringify(status))))
+						);
+					});
+					describe('And the WFH contains just an emoji', () => {
+						beforeEach(() => {
+							text = randomEmoji;
+							status.status_emoji = ':'.concat(emoji.which(randomEmoji), ':');
+							return act();
+						});
+						it('should set status emoji to specified and status message to default \'Working from home\'', () =>
+							sinon.assert.calledWith(fakeSlackApi.sendResponse, 'slack.com/api/users.profile.set', 'token='.concat(token, '&profile=', encodeURIComponent(JSON.stringify(status))))
+						);
+					});
+					describe('And the WFH contains multiple emojis', () => {
+						beforeEach(() => {
+							status.status_text = randomEmoji.concat(text);
+							status.status_emoji = ':'.concat(emoji.which(randomEmoji), ':');
+							text = randomEmoji.concat(randomEmoji, text);
+							return act();
+						});
+						it('should set status emoji to first emoji and status message to remainder of text', () =>
+							sinon.assert.calledWith(fakeSlackApi.sendResponse, 'slack.com/api/users.profile.set', 'token='.concat(token, '&profile=', encodeURIComponent(JSON.stringify(status))))
+						);
+					});
+					describe('And the WFH contains the status message before the emoji', () => {
```

Comment:
You really thought about all the different use cases here. That is fantastic. I can't really think of any test cases you have missed. I really like the 2 emoji and emoji at the end of the status test cases. I hadn't even though of those, so that's great. Good work thinking through all the scenarios. 👍